include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -2934,6 +2934,347 @@ namespace Differentiation
+     *     // These could happily be set to anything, unless the function will
+     *     // be evaluated along a branch not otherwise traversed during later
+     *     // use. For this reason, in this example instead of using some dummy
+     *     // values, we'll actually map out the function at the same point
+     *     // around which we'll later linearize it.
+     *     ad_helper.register_independent_variable(H, H_dofs);
+     *     ad_helper.register_independent_variable(C, C_dofs);
+     *
+     *     // NOTE: We have to extract the sensitivities in the order we wish to
+     *     // introduce them. So this means we have to do it by logical order
+     *     // of the extractors that we've created.
+     *     const SymmetricTensor<2,dim,ADNumberType> C_AD =
+     *       ad_helper.get_sensitive_variables(C_dofs); const
+     *     const Tensor<1,dim,ADNumberType>          H_AD =
+     *       ad_helper.get_sensitive_variables(H_dofs);
+     *
+     *     // Here we define the material stored energy function.
+     *     // This example is sufficiently complex to warrant the use of AD to,
+     *     // at the very least, verify an unassisted implementation.
+     *     const double mu_e = 10;          // Shear modulus
+     *     const double lambda_e = 15;      // Lame parameter
+     *     const double mu_0 = 4*M_PI*1e-7; // Magnetic permeability constant
+     *     const double mu_r = 5;           // Relative magnetic permeability
+     *
+     *     const ADNumberType J = std::sqrt(determinant(C_AD));
+     *     const SymmetricTensor<2,dim,ADNumberType> C_inv_AD = invert(C_AD);
+     *     const ADNumberType psi =
+     *       0.5*mu_e*(1.0+std::tanh((H_AD*H_AD)/100.0))*
+     *         (trace(C_AD) - dim - 2*std::log(J)) +
+     *       lambda_e*std::log(J)*std::log(J) -
+     *       0.5*mu_0*mu_r*J*H_AD*C_inv_AD*H_AD;
+     *
+     *     // Register the definition of the total stored energy
+     *     ad_helper.register_dependent_variable(psi_CH);
+     *
+     *     // Indicate that we have completed tracing the operations onto
+     *     // the tape.
+     *     ad_helper.stop_recording_operations(false); // write_tapes_to_file
+     *   }
+     *   else
+     *   {
+     *     // This is the "tape reuse" phase of the operations.
+     *     // Here we will leverage the already traced operations that reside
+     *     // on a tape, and simply re-evaluate the tape at a different point
+     *     // to get the function values and their derivatives.
+     *
+     *     // Load the existing tape to be reused
+     *     ad_helper.activate_recorded_tape(tape_index);
+     *
+     *     // Set the new values of the independent variables where the
+     *     // recorded dependent functions are to be evaluated (and
+     *     // differentiated around).
+     *     ad_helper.set_independent_variable(C, C_dofs);
+     *     ad_helper.set_independent_variable(H, H_dofs);
```

Comment:
Out of curiosity, is this the place where you would have to consider re-taping if there were a branch somewhere?

Reply:
(I'm feeling quite proud of myself here because I'm thinking that I'm actually asking a meaningful question. But that might simply be a reflection of how little I really understand about all this...)

Reply:
> Out of curiosity, is this the place where you would have to consider re-taping if there were a branch somewhere?

Yes, exactly! Setting these variables could lead to a branch switch, but if my memory serves me correctly then it'll only be triggered when `compute_[value/gradient/hessian]()` is called. I should check this out sometime.